src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/LanguageServices/VisualBasicSyntaxFactsService.vb

```diff
@@ -1410,6 +1410,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return False
         End Function
 
+        Public Function IsNamespaceDeclaration(node As SyntaxNode) As Boolean Implements ISyntaxFactsService.IsNamespaceDeclaration
+            Select Case node.Kind()
+                Case SyntaxKind.NamespaceBlock,
+                     SyntaxKind.NamespaceStatement
```

Comment:
Why the block _and_ the namespace?

Reply:
`IsDeclaration` includes both. It looks like `IsTypeDeclaration` doesn't include statements, only blocks though.